Modules/Core/SpatialObjects/include/itkSpatialObjectPoint.h

```diff
@@ -45,25 +51,27 @@ class ITK_TEMPLATE_EXPORT SpatialObjectPoint
   virtual ~SpatialObjectPoint() = default;
 
   using Self = SpatialObjectPoint;
+
+  using SpatialObjectType = SpatialObject<TPointDimension>;
+
   using PointType = Point< double, TPointDimension >;
   using VectorType = vnl_vector< double >;
-  using PixelType = RGBAPixel< float >;
-  using ColorType = PixelType;
-
-  /** Get the SpatialObjectPoint Id. */
-  int GetID() const;
+  using ColorType = RGBAPixel< double >;
 
   /** Set the SpatialObjectPoint Id. */
-  void SetID(const int newID);
+  void SetId( int id )
+  { m_Id = id; }
 
-  /** Return a pointer to the point object. */
-  const PointType & GetPosition() const;
+  /** Get the SpatialObjectPoint Id. */
+  int GetId() const
+  { return m_Id; }
 
-  /** Set the point object. Couldn't use macros for these methods. */
-  void SetPosition(const PointType & newX);
+  /** Set the point object. */
+  void SetPositionInObjectSpace(const PointType & newPositionInObjectSpace)
+  { m_PositionInObjectSpace = newPositionInObjectSpace; }
```

Comment:
Shouldn't this function call `Modified` if the new value is different from the old one?

Reply:
Thanks!  Should use the ITK macro.   I'll make another pass through fixing these Set (and Get) calls that are paired to member variables.

Reply:
Actually, a "point" is a lightweight data container.  It is not an ITK-derived class, so it is not using Update/Modified or ITK macros.   I had thought about deriving it from an ITK point, but none of the API overlaps.  I think it is ok for this to not have Modified/Update calls, but I'm open to other suggestions.

Reply:
Indeed. I hadn't realized that this SpatialObjectPoint class was different. I am not familiar enough with where it is used to know if it should be changed. It really depends if changing the value of the point should lead to updating ITK pipelines in which it is used. I'll let you make that decision.

Reply:
I consider SpatialObjectPoints to be relatively static data items.   They are typically added to or removed from a pointbasedspatialobject - which causes that spatialobject's Modified() function to be called, so I do think we're doing the right thing as-is.